### PR TITLE
Fix Rancher monitoring deployment errors

### PR DESCRIPTION
**Pull request by moio**
Previous code did not wait after Rancher installation to install Rancher Monitoring.

Since 2.9, some Rancher components need to be ready or the installation will fail, eg.:

```
2024/08/29 14:50:01 chart rancher-monitoring: failed pre-install: warning: Hook pre-install rancher-monitoring/templates/alertmanager/secret.yaml failed: 1 error occurred:
        * Internal error occurred: failed calling webhook "rancher.cattle.io.secrets": failed to call webhook: Post "https://rancher-webhook.cattle-system.svc:443/v1/webhook/mutation/secrets?timeout=15s": no endpoints available fo
r service "rancher-webhook"
```

More components can be added if needed in the same way (eg. `cattle-fleet-system/gitjob` or `cattle-provisioning-capi-system/capi-controller-manager`, which did not pop up in my own tests but might turn out to be relevant).

### CODE REVIEW

**Comment by git-ival** [comment]
LGTM, waiting on #6 to be merged before approving + merging

**Comment by moio** [comment]
Rebased on main, PR is now clean from extraneous commits

**Review by git-ival — APPROVED** [review]

**Pull request merged**
